infrasim/ipmiconsole/common.py

```diff
@@ -79,10 +79,19 @@ def init_env(instance):
     cur_path = os.environ["PATH"]
     os.environ["PATH"] = "{}/bin:{}".format(os.environ.get("PYTHONPATH"), cur_path)
     if not Workspace.check_workspace_exists(instance):
-        raise IpmiError("Warning: there is no node {} workspace. Please start node {} first.".format(instance, instance))
-    output = run_command("infrasim node status")
-    if output[0] == 0 and "{}-bmc is stopped".format(instance) in output[1]:
-        raise IpmiError("Warning: node {} has not started BMC. Please start node {} first.".format(instance, instance))
+        raise IpmiError(
+            "Warning: there is no node {} workspace. "
+            "Please start node {} first.".format(instance, instance))
+    try:
+        with open("{}/{}/.{}-bmc.pid".format(
+                config.infrasim_home, instance, instance), "r") as f:
+            pid = f.readline().strip()
+            if not os.path.exists("/proc/{}".format(pid)):
+                raise Exception
+    except Exception:
+        raise IpmiError(
+            "Warning: node {} has not started BMC. "
+            "Please start node {} first.".format(instance, instance))
```

Comment:
Deserve functional tests on:
- Stop node and verify ipmi-console is stopped as well
- Exception to start ipmi-console for a stopped node